nupic/research/spatial_pooler.py

```diff
@@ -42,6 +43,57 @@ class InvalidSPParamValueError(ValueError):
 
 
 
+class _SparseMatrixCorticalColumnAdapter(object):
+  """ Many functions in SpatialPooler operate on a columnIndex but use an
+  underlying storage implementation based on a Sparse Matrix in which cortical
+  columns are represented as rows.  This can be confusing to someone trying to
+  follow the algorithm, confusing terminology between matrix math and cortical
+  columns.  This class is provided to abstract away some of the details of the
```

Comment:
How would you do that?

Only way I can think of is to add some sort of hack in `__new__()` or `__init__()` to check `__bases__` or compare `self.__class__`.

You could make `__init__()` an abstract method, which would result in a failure if someone attempts to instantiate it directly.

Instead, maybe better to assert int `__init__()` that `super(_SparseMatrixCorticalColumnAdapter, self)` has the attrs that it expects, like `getRows`, `replaceSparseRow`, etc.?